opengrok-indexer/src/main/java/org/opengrok/indexer/framework/PluginFramework.java

```diff
@@ -313,9 +319,24 @@ private String getClassName(File f) {
         return classname;
     }
 
+    @Nullable
     private String getClassName(JarEntry f) {
+        final String filePath = f.getName();
+        if (!filePath.endsWith(CLASS_SUFFIX)) {
+            return null;
+        }
+
+        File file = new File(pluginDirectory.getAbsolutePath(), filePath);
+        try {
+            if (!file.getCanonicalPath().startsWith(pluginDirectory.getCanonicalPath())) {
```

Comment:
Shouldn't the `File.separator` be appended?
```java
if (!file.getCanonicalPath().startsWith(pluginDirectory.getCanonicalPath() + File.separator)) {
```
At least the example on https://snyk.io/research/zip-slip-vulnerability has it.

Reply:
Good point.